test/Spec/GT/DB/Usage.hs

```diff
@@ -1,45 +1,58 @@
+import GT.DB.Schema.Migration
+import GT.DB.Schema.Ride
+import GT.DB.Usage            as Usage
 
-spec :: Spec
+spec :: SpecWith TestDb
 spec = do
     describe "new" $ do
-        it "works" $ do
-            pendingWith "really need a better database testing story!!"
-            usageIds <- runNoLoggingT $
-                withPostgresqlConn "dbname=geartracker" $ \conn -> do
-                    flip runSqlConn conn $ do
-                        componentId <- insert Component
-                            { componentName = "Scrubtrout"
-                            , componentPart = "Frame"
-                            , componentBrand = "Salsa"
-                            , componentModel = "Cutthroat"
-                            , componentDescription = ""
-                            , componentParent = Nothing
-                            }
-                        bikeId <- insert $ Bike componentId
-                        wheelId <- insert Component
-                            { componentName = "Good wheel"
-                            , componentPart = "Front wheel"
-                            , componentBrand = "WTB"
-                            , componentModel = "Asym 29"
-                            , componentDescription = "it is p good"
-                            , componentParent = Just componentId
-                            }
-                        rideId <- insert Ride
-                            { rideDistance = 32
-                            , rideDate = UTCTime (fromGregorian 2020 01 03) 0
-                            }
+        itDb "works" $ do
+            frameId <- insert Component
+                { componentName = "Scrubtrout"
+                , componentPart = "Frame"
+                , componentBrand = "Salsa"
+                , componentModel = "Cutthroat"
+                , componentDescription = ""
+                , componentParent = Nothing
+                }
+            bikeId <- insert $ Bike frameId
+            wheelId <- insert Component
+                { componentName = "Good wheel"
+                , componentPart = "Front wheel"
+                , componentBrand = "WTB"
+                , componentModel = "Asym 29"
+                , componentDescription = "it is p good"
+                , componentParent = Just frameId
+                }
+            let tire = Component
+                    { componentName = "knobbss"
+                    , componentPart = "Tire"
+                    , componentBrand = "WTB"
+                    , componentModel = "Riddler 45"
+                    , componentDescription = ""
+                    , componentParent = Just wheelId
+                    }
+            tireId <- insert tire
+            rideId <- insert Ride
+                { rideDistance = 32
+                , rideDate = UTCTime (fromGregorian 2020 01 03) 0
+                }
 
-                        Usage.new bikeId rideId
-            length usageIds `shouldBe` 2
```

Comment:
This is such a bad test! It was succeeding correctly with `[UsageKey 1, UsageKey 1]`, because we're not measuring what we actually care about.